lib/rspec/mocks/error_generator.rb

```diff
@@ -291,6 +303,19 @@ def format_received_args(*args)
         args.empty? ? "(no args)" : "(" + received_arg_list(*args) + ")"
       end
 
+      def format_unexpected_with_args(*args)
+        with_args = Hash[Array(args.first).group_by { |x| x }.map { |k, v| [k, v.count] }]
```

Comment:
Why are you passing `args.first` to `Array`?  I thought that `args` is a list of arg lists (one per method call), so this looks to me like it just is the arg list from the first call...but that doesn't make sense and it works so there's something I'm not understanding here.